ts/features/euCovidCert/screens/EuCovidCertificateRouterScreen.tsx

```diff
@@ -99,16 +103,34 @@ const EuCovidCertificateRouterScreen = (
   }, [setMessageRead, shouldBeLoaded, loadCertificate, messageId, authCode]);
 
   // handle with the fold the remote state and with routeEuCovidResponse the different response values
-  return pot.fold(
-    props.euCovidCertificateResponse(authCode),
-    () => <EuCovidCertLoadingScreen />,
-    () => <EuCovidCertLoadingScreen />,
-    _ => <EuCovidCertLoadingScreen />,
-    _ => <EuCovidCertGenericErrorKoScreen />,
-    response => routeEuCovidResponse(response),
-    _ => <EuCovidCertLoadingScreen />,
-    (_, __) => <EuCovidCertLoadingScreen />,
-    _ => <EuCovidCertGenericErrorKoScreen />
+  return (
+    <EUCovidContext.Provider value={{ authCode, messageId }}>
+      {pot.fold(
+        props.euCovidCertificateResponse(authCode),
+        () => (
+          <EuCovidCertLoadingScreen />
+        ),
+        () => (
+          <EuCovidCertLoadingScreen />
+        ),
+        _ => (
+          <EuCovidCertLoadingScreen />
+        ),
+        _ => (
+          <EuCovidCertGenericErrorKoScreen />
+        ),
+        response => routeEuCovidResponse(response),
+        _ => (
+          <EuCovidCertLoadingScreen />
+        ),
+        (_, __) => (
+          <EuCovidCertLoadingScreen />
+        ),
+        _ => (
+          <EuCovidCertGenericErrorKoScreen />
+        )
+      )}
+    </EUCovidContext.Provider>
```

Comment:
In this case I think is better to keep the explicit folding, mainly for the clearity that the IDE give about each case:
<img width="453" alt="Schermata 2021-11-08 alle 10 53 44" src="https://user-images.githubusercontent.com/26501317/140721043-936f9f90-0019-41f4-ab2e-ceea8890014a.png">

Otherwise to help is given and we need to remember each fold argument which case it belongs to:
<img width="382" alt="Schermata 2021-11-08 alle 10 54 54" src="https://user-images.githubusercontent.com/26501317/140721275-eb19e24b-a020-4ecc-8999-9ceabfff1625.png">